test/compliance/SELECT.test.js

```diff
@@ -635,6 +635,20 @@ describe('SELECT', () => {
       const res = await cds.run(cqn)
       assert.strictEqual(res.length, 3, 'Ensure that all rows are coming back')
     })
+
+    test('navigation with duplicate identifier in path', async () => {
+      const { Books } = cds.entities('complex.associations')
+      const cqn = CQL`SELECT name { name } FROM ${Books} GROUP BY name.name`
+      const res = await cds.run(cqn)
+      assert.strictEqual(res.length, 1, 'Ensure that all rows are coming back')
+    })
```

Comment:
Just an heads up there is a new API for `CQL` which is `cds.ql` which can be executed directly making the test look like:

```suggestion
    test('navigation with duplicate identifier in path', async () => {
      const { Books } = cds.entities('complex.associations')
      const res = await cds.ql`SELECT name { name } FROM ${Books} GROUP BY name.name`
      assert.strictEqual(res.length, 1, 'Ensure that all rows are coming back')
    })
```